### PR TITLE
create types for hummer-env

### PR DESCRIPTION
**Pull request by iammvp**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by peterblazejewicz** [comment]
Hey, what are those packages? Are those 'hummer/*' or '@hummer/*' (namespaced). Are those published to NPM?
Are those exports for @hummer/tenon-*? those seem to have it's own declaration being native TS:
https://github.com/didi/Hummer/blob/master/tenon/packages/tenon-core/tsconfig.json#L9

File: types/@hummer/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,248 @@
+// Type definitions for hummer 1.0
```
Comment (sandersn) [inline types/@hummer/index.d.ts:1]: the directory name should be `hummer` not `@hummer` to match the package name.

File: types/@hummer/@hummer-tests.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import Hummer from './emulator/Hummer';
```
Comment (sandersn) [inline types/@hummer/@hummer-tests.ts:1]: 1. filename should be hummer-tests.ts
2. It looks like only ./emulation/Hummer has exports, and the rest of the package is global. But the tests should test as much as possible, so I'd recommend testing the global `Hummer` object with no imports.

**Review by sandersn — CHANGES_REQUESTED** [review]

**Comment by iammvp** [comment]
> Hey, what are those packages? Are those 'hummer/_' or '@hummer/_' (namespaced). Are those published to NPM? Are those exports for @hummer/tenon-*? those seem to have it's own declaration being native TS: https://github.com/didi/Hummer/blob/master/tenon/packages/tenon-core/tsconfig.json#L9

@peterblazejewicz Thanks for reply me. 

1. This type definition is for '@hummer'
2. '@hummer' will not publish to NPM because it's a native(iOS/Android) render library which can use JS/TS to render the native page. And '@hummer' inject serval global APIs to JS engine, this PR is for those APIs.
3. '@hummer/tenon-*' is a sub-library for '@hummer' ecosystem,  but it doesn't cover those global APIs definitions.
4. I've contacted '@hummer' maintainers, they are happy that I can define it for them.

File: types/@hummer/index.d.ts (hunk 1 above)
Comment (iammvp) [reply types/@hummer/index.d.ts:1]: Hi sandersn, as mentioned in a previous comment, this is a definition for '@hummer' not 'hummer', please check the previous comment for more details.

File: types/@hummer/@hummer-tests.ts (hunk 2 above)
Comment (iammvp) [reply types/@hummer/@hummer-tests.ts:1]: Hummer is a global API injected by an iOS/Android render library, so there is no such context to test it on Node.js environment. I am wondering how to test those APIs without an environment. Could you help me to figure out the proper way?

**Review by sandersn — CHANGES_REQUESTED** [review]

**Review by sandersn — APPROVED** [review]

**Pull request merged**